Test failure prior to fix (revert fix commit and check):

```console
❯ bench run-tests --doctype "Company"
❯ bench run-tests --doctype "Customer"
❯ bench run-tests --module "erpnext.accounts.doctype.loyalty_program.test_loyalty_program"
/nix/store/srn5qp23ili0a1gbq2q8v2yqyshqfkws-python3-3.11.9-env/lib/python3.11/site-packages/passlib/utils/__init__.py:854: DeprecationWarning: 'crypt' is deprecated and slated for removal in Python 3.13
  from crypt import crypt as _crypt

test_cancel_sales_invoice (erpnext.accounts.doctype.loyalty_program.test_loyalty_program.TestLoyaltyProgram.test_cancel_sales_invoice)
cancelling the sales invoice should cancel the earned points (2.21s)
.....FFF
======================================================================
FAIL: test_tier_selection (erpnext.accounts.doctype.loyalty_program.test_loyalty_program.TestLoyaltyProgram.test_tier_selection) (total_spent=0, current_transaction_amount=6000)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/blaggacao/src/gitlab.com/pristina/infraestructura/apps/erpnext/erpnext/accounts/doctype/loyalty_program/test_loyalty_program.py", line 255, in test_tier_selection
    self.assertEqual(
AssertionError: 'Silver' != 'Bronze'
- Silver
+ Bronze
 : Expected tier Bronze for total_spent 0 and current_transaction_amount 6000, but got Silver

======================================================================
FAIL: test_tier_selection (erpnext.accounts.doctype.loyalty_program.test_loyalty_program.TestLoyaltyProgram.test_tier_selection) (total_spent=0, current_transaction_amount=15000)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/blaggacao/src/gitlab.com/pristina/infraestructura/apps/erpnext/erpnext/accounts/doctype/loyalty_program/test_loyalty_program.py", line 255, in test_tier_selection
    self.assertEqual(
AssertionError: 'Gold' != 'Silver'
- Gold
+ Silver
 : Expected tier Silver for total_spent 0 and current_transaction_amount 15000, but got Gold

======================================================================
FAIL: test_tier_selection (erpnext.accounts.doctype.loyalty_program.test_loyalty_program.TestLoyaltyProgram.test_tier_selection) (total_spent=8000, current_transaction_amount=3000)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/blaggacao/src/gitlab.com/pristina/infraestructura/apps/erpnext/erpnext/accounts/doctype/loyalty_program/test_loyalty_program.py", line 255, in test_tier_selection
    self.assertEqual(
AssertionError: 'Gold' != 'Silver'
- Gold
+ Silver
 : Expected tier Silver for total_spent 8000 and current_transaction_amount 3000, but got Gold

----------------------------------------------------------------------
Ran 6 tests in 5.229s

FAILED (failures=3)
```

--- a/erpnext/accounts/doctype/loyalty_program/loyalty_program.py
+++ b/erpnext/accounts/doctype/loyalty_program/loyalty_program.py
@@ -86,10 +86,9 @@ def get_loyalty_program_details_with_points(
 	tier_spent_level = sorted(
 		[d.as_dict() for d in loyalty_program.collection_rules],
 		key=lambda rule: rule.min_spent,
-		reverse=True,
 	)
 	for i, d in enumerate(tier_spent_level):
-		if i == 0 or (lp_details.total_spent + current_transaction_amount) <= d.min_spent:
+		if i == 0 or (lp_details.total_spent + current_transaction_amount) >= d.min_spent:
 			lp_details.tier_name = d.tier_name
 			lp_details.collection_factor = d.collection_factor
 		else:

--- a/erpnext/accounts/doctype/loyalty_program/loyalty_program.py
+++ b/erpnext/accounts/doctype/loyalty_program/loyalty_program.py
@@ -79,9 +79,10 @@ def get_loyalty_program_details_with_points(
 ):
 	lp_details = get_loyalty_program_details(customer, loyalty_program, company=company, silent=silent)
 	loyalty_program = frappe.get_doc("Loyalty Program", loyalty_program)
-	lp_details.update(
-		get_loyalty_details(customer, loyalty_program.name, expiry_date, company, include_expired_entry)
+	loyalty_details = get_loyalty_details(
+		customer, loyalty_program.name, expiry_date, company, include_expired_entry
 	)
+	lp_details.update(loyalty_details)
 
 	tier_spent_level = sorted(
 		[d.as_dict() for d in loyalty_program.collection_rules],

--- a/erpnext/accounts/doctype/loyalty_program/test_loyalty_program.py
+++ b/erpnext/accounts/doctype/loyalty_program/test_loyalty_program.py
@@ -7,6 +7,7 @@ import frappe
 from frappe.utils import cint, flt, getdate, today
 
 from erpnext.accounts.doctype.loyalty_program.loyalty_program import (
+	get_loyalty_details,
 	get_loyalty_program_details_with_points,
 )
 from erpnext.accounts.party import get_dashboard_info
@@ -196,6 +197,70 @@ class TestLoyaltyProgram(unittest.TestCase):
 
 		for d in company_wise_info:
 			self.assertTrue(d.get("loyalty_points"))
+
+	@unittest.mock.patch("erpnext.accounts.doctype.loyalty_program.loyalty_program.get_loyalty_details")
+	def test_tier_selection(self, mock_get_loyalty_details):
+		# Create a new loyalty program with multiple tiers
+		loyalty_program = frappe.get_doc(
+			{
+				"doctype": "Loyalty Program",
+				"loyalty_program_name": "Test Tier Selection",
+				"auto_opt_in": 1,
+				"from_date": today(),
+				"loyalty_program_type": "Multiple Tier Program",
+				"conversion_factor": 1,
+				"expiry_duration": 10,
+				"company": "_Test Company",
+				"cost_center": "Main - _TC",
+				"expense_account": "Loyalty - _TC",
+				"collection_rules": [
+					{"tier_name": "Gold", "collection_factor": 1000, "min_spent": 20000},
+					{"tier_name": "Silver", "collection_factor": 1000, "min_spent": 10000},
+					{"tier_name": "Bronze", "collection_factor": 1000, "min_spent": 5000},
+				],
+			}
+		)
+		loyalty_program.insert()
+
+		# Test cases with different total_spent and current_transaction_amount combinations
+		test_cases = [
+			(0, 6000, "Bronze"),
+			(0, 15000, "Silver"),
+			(0, 25000, "Gold"),
+			(4000, 500, "Bronze"),
+			(8000, 3000, "Silver"),
+			(18000, 3000, "Gold"),
+			(22000, 5000, "Gold"),
+		]
+
+		for total_spent, current_transaction_amount, expected_tier in test_cases:
+			with self.subTest(total_spent=total_spent, current_transaction_amount=current_transaction_amount):
+				# Mock the get_loyalty_details function to update the total_spent
+				def side_effect(*args, **kwargs):
+					result = get_loyalty_details(*args, **kwargs)
+					result.update({"total_spent": total_spent})
+					return result
+
+				mock_get_loyalty_details.side_effect = side_effect
+
+				lp_details = get_loyalty_program_details_with_points(
+					"Test Loyalty Customer",
+					loyalty_program=loyalty_program.name,
+					company="_Test Company",
+					current_transaction_amount=current_transaction_amount,
+				)
+
+				# Get the selected tier based on the current implementation
+				selected_tier = lp_details.tier_name
+
+				self.assertEqual(
+					selected_tier,
+					expected_tier,
+					f"Expected tier {expected_tier} for total_spent {total_spent} and current_transaction_amount {current_transaction_amount}, but got {selected_tier}",
+				)
+
+		# Clean up
+		loyalty_program.delete()
 
 
 def get_points_earned(self):

--- a/erpnext/accounts/doctype/loyalty_program_collection/loyalty_program_collection.json
+++ b/erpnext/accounts/doctype/loyalty_program_collection/loyalty_program_collection.json
@@ -20,6 +20,7 @@
   {
    "fieldname": "min_spent",
    "fieldtype": "Currency",
+   "in_list_view": 1,
    "label": "Minimum Total Spent"
   },
   {
@@ -37,7 +38,7 @@
  ],
  "istable": 1,
  "links": [],
- "modified": "2024-03-27 13:10:03.536071",
+ "modified": "2024-09-05 07:41:25.694041",
  "modified_by": "Administrator",
  "module": "Accounts",
  "name": "Loyalty Program Collection",

